frontend/src/scenes/cohorts/Cohort.tsx

```diff
@@ -1,199 +1,300 @@
-import { CohortNameInput } from './CohortNameInput'
-import { CohortDescriptionInput } from './CohortDescriptionInput'
-import { Col, Divider, Row, Tooltip } from 'antd'
-import { CohortMatchingCriteriaSection } from './CohortMatchingCriteriaSection'
-import { AvailableFeature, CohortType } from '~/types'
-import { COHORT_DYNAMIC, COHORT_STATIC } from 'lib/constants'
-import { InboxOutlined, CalculatorOutlined, OrderedListOutlined } from '@ant-design/icons'
+import { Group, Field as KeaField } from 'kea-forms'
+import { Col, Divider, Row } from 'antd'
+import { AvailableFeature, CohortGroupType, CohortType } from '~/types'
+import { CohortTypeType, ENTITY_MATCH_TYPE, PROPERTY_MATCH_TYPE } from 'lib/constants'
+import { CalculatorOutlined, OrderedListOutlined, PlusOutlined } from '@ant-design/icons'
 import Dragger from 'antd/lib/upload/Dragger'
-import { CohortDetailsRow } from './CohortDetailsRow'
 import { Persons } from 'scenes/persons/Persons'
 import { cohortLogic } from './cohortLogic'
-import { UploadFile } from 'antd/lib/upload/interface'
-import { DropdownSelector } from 'lib/components/DropdownSelector/DropdownSelector'
 import { userLogic } from 'scenes/userLogic'
 import 'antd/lib/dropdown/style/index.css'
 import { Spinner } from 'lib/components/Spinner/Spinner'
 import { SceneExport } from 'scenes/sceneTypes'
 import { LemonButton } from 'lib/components/LemonButton'
 import { PageHeader } from 'lib/components/PageHeader'
+import { LemonInput } from 'lib/components/LemonInput'
+import { Field } from 'lib/forms/Field'
+import { VerticalForm } from 'lib/forms/VerticalForm'
+import { LemonSelect, LemonSelectOptions } from 'lib/components/LemonSelect'
+import { LemonTextArea } from 'lib/components/LemonTextArea'
+import { IconUploadFile } from 'lib/components/icons'
+import { UploadFile } from 'antd/es/upload/interface'
+import { MatchCriteriaSelector } from 'scenes/cohorts/MatchCriteriaSelector'
 
 export const scene: SceneExport = {
     component: Cohort,
     logic: cohortLogic,
-    paramsToProps: ({ params: { id } }) => ({ id: id && id !== 'new' ? parseInt(id) : 'new', pageKey: 0 }),
+    paramsToProps: ({ params: { id } }) => ({ id: id && id !== 'new' ? parseInt(id) : 'new' }),
 }
 
-let uniqueMemoizedIndex = 0
+const COHORT_TYPE_OPTIONS: LemonSelectOptions = {
+    [CohortTypeType.Static]: {
+        label: 'Static · Updated manually',
+        icon: <OrderedListOutlined />,
+    },
+    [CohortTypeType.Dynamic]: {
+        label: 'Dynamic · Updates automatically',
+        icon: <CalculatorOutlined />,
+    },
+}
 
 export function Cohort({ id }: { id?: CohortType['id'] } = {}): JSX.Element {
-    const pageKey = useMemo(() => uniqueMemoizedIndex++, [id])
-    const logic = cohortLogic({ pageKey, id })
-    const { setCohort, saveCohort, deleteCohort } = useActions(logic)
-    const { cohort, submitted } = useValues(logic)
+    const logicProps = { id }
+    const logic = cohortLogic(logicProps)
```

Comment:
Not super familiar here but is there no need for memoized pageKey anymore?

Reply:
Yea this was causing a bug where two differently keyed logics were being mounted, which caused data discrepancies